File: Contest 1692/4.py
answers = []

for _ in range(int(input())):
    s, x = input().split()
    x = int(x)

    s = s.split(":")
    h = int(s[0])
    m = int(s[1])

    ih, im = h, m
    ma = x % 60
    ha = x // 60

    started = False
    palindromes = set()
    while not (h == ih and m == im) or not started:
        sh = list(str(h))
        if h < 10:
            sh = ['0'] + sh
        sh.reverse()
        sm = list(str(m))
        if m < 10:
            sm = ['0'] + sm
        if sh == sm:
            sh.reverse()
            palindromes.add(f"{sh} {sm}")
        h += ha
        m += ma
        if h > 23:
            h %= 24
        if m > 59:
            h += 1
            m = m % 60
            if h == 24:
                h = 0
        started = True

    answers.append(len(palindromes))

for ans in answers:
    print(ans)

File: Contest 1692/4.cpp
#include <bits/stdc++.h>
#define endl "\n"
typedef long long ll;
using namespace std;

void solve()
{
    string s;
    int x;
    cin >> s >> x;

    int h = (s[0] - '0') * 10 + (s[1] - '0');
    int m = (s[3] - '0') * 10 + (s[4] - '0');
    int ih = h, im = m;
    int ha = x / 60;
    int ma = x % 60;

    bool started = false;
    set<pair<string, string>> palindromes;
    while (!(ih == h && im == m) || !started) {
        string sh = to_string(h);
        string sm = to_string(m);

        if (sh.length() == 1)
            sh = "0" + sh;
        if (sm.length() == 1)
            sm = "0" + sm;

        string r_sm = sm;
        reverse(r_sm.begin(), r_sm.end());
        if (sh == r_sm) {
            palindromes.insert({sh, sm});
        }

        h += ha, m += ma;
        if (h > 23) {
            h %= 24;
        }
        if (m > 59) {
            h += 1;
            m %= 60;
            if (h == 24) {
                h = 0;
            }
        }
        started = true;
    }

    cout << palindromes.size() << endl;
}

int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t;
    cin >> t;
    while (t--) {
        solve();
    }
}